collector/hyperv.go

```diff
@@ -0,0 +1,502 @@
+// returns data points from 
+// Win32_PerfRawData_HvStats_HyperVHypervisor
+// Win32_PerfRawData_HvStats_HyperVHypervisorLogicalProcessor
+
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+
+func init() {
+	Factories["hyperv"] = NewHypervCollector
+}
+
+
+// A HypervCollector is a Prometheus collector for WMI
+// Win32_PerfRawData_HvStats_HyperVHypervisorLogicalProcessor
+// and Win32_PerfRawData_HvStats_HyperVHypervisor metrics.
+type HypervCollector struct {
+	// Win32_PerfRawData_HvStats_HyperVHypervisor	
+	LogicalProcessors					*prometheus.Desc
+	MonitoredNotifications				*prometheus.Desc
+	Partitions							*prometheus.Desc
+	TotalPages							*prometheus.Desc
+	VirtualProcessors					*prometheus.Desc
+
+	//Win32_PerfRawData_HvStats_HyperVHypervisorLogicalProcessor
+	C1TransitionsPersec					*prometheus.Desc
+	C2TransitionsPersec					*prometheus.Desc
+	C3TransitionsPersec					*prometheus.Desc
+	ContextSwitchesPersec				*prometheus.Desc
+	Frequency							*prometheus.Desc
+	HardwareInterruptsPersec			*prometheus.Desc
+	InterProcessorInterruptsPersec		*prometheus.Desc
+	InterProcessorInterruptsSentPersec	*prometheus.Desc
+	MonitorTransitionCost				*prometheus.Desc
+	ParkingStatus						*prometheus.Desc
+	PercentC1Time						*prometheus.Desc
+	PercentC2Time						*prometheus.Desc
+	PercentC3Time						*prometheus.Desc
+	PercentGuestRunTime					*prometheus.Desc
+	PercentHypervisorRunTime			*prometheus.Desc
+	PercentIdleTime						*prometheus.Desc
+	PercentofMaxFrequency				*prometheus.Desc
+	PercentTotalRunTime					*prometheus.Desc
+	ProcessorStateFlags					*prometheus.Desc
+	RootVpIndex							*prometheus.Desc
+	SchedulerInterruptsPersec			*prometheus.Desc
+	TimerInterruptsPersec				*prometheus.Desc
+	TotalInterruptsPersec				*prometheus.Desc
+}
+
+
+func NewHypervCollector() (Collector, error) {
+	const subsystem = "hypervisor"
+
+	//func NewDesc(fqName, help string, variableLabels []string, constLabels Labels) *Desc
+	//func BuildFQName(namespace, subsystem, name string) string
+
+	return &HypervCollector{
+		LogicalProcessors: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "Logical_Processors_Number"),
```

Comment:
All names should be lower-case. Also make sure to follow the guidelines [here](https://prometheus.io/docs/instrumenting/writing_exporters/) for naming.